numpy/core/src/umath/loops_arithmetic.dispatch.c.src

```diff
@@ -56,18 +154,30 @@ simd_divide_by_scalar_contig_@sfx@(char **args, npy_intp len)
 
 /**begin repeat
  * Unsigned types
- *  #type  = npy_ubyte, npy_ushort, npy_uint, npy_ulong, npy_ulonglong#
- *  #TYPE  = UBYTE,     USHORT,     UINT,     ULONG,     ULONGLONG#
- *  #STYPE = BYTE,      SHORT,      INT,      LONG,      LONGLONG#
+ *  #type  = byte, short, int, long, longlong#
+ *  #TYPE  = BYTE, SHORT, INT, LONG, LONGLONG#
  */
+
+/**begin repeat1
+ * #signed = 1, 0#
+ */
+
 #undef TO_SIMD_SFX
+#undef SIMD_TYPE
+#undef SIMD_DIVIDE
 #if 0
-/**begin repeat1
+/**begin repeat2
  * #len = 8, 16, 32, 64#
  */
-#elif NPY_BITSOF_@STYPE@ == @len@
+#elif NPY_BITSOF_@TYPE@ == @len@ && @signed@
```

Comment:
Having a separated loop for signed division will be much easier to read, and allow keeping the previous macro `TO_SIMD_SFX`.